com.unity.sg2/Editor/GraphDeltaRegistry/Generation/Interpreter.cs

```diff
@@ -20,6 +21,573 @@ namespace UnityEditor.ShaderGraph.Generation
+                var inputType = inputBuilder.Build();
+                typeReferences.Add(inputType);
+                mainFunction.AddInput(inputType, "In");
+                return new ProcessingResult() { entryPointFunction = mainFunction.Build(), functions = functionReferences.Values, includes = includeReferences, usedTypes = typeReferences };
+            }
+
+            private void WriteFunctionCall(FunctionReference functionRef, IEnumerable<VariableReference> parameterReferences)
+            {
+                List<string> paramStrings = new List<string>();
+                foreach(var paramRef in parameterReferences)
+                {
+                    var param = variableReferences[paramRef];
+                    if (!param.inlineInitialization)
+                    {
+                        var s = param.actualVariableName ?? param.desiredVariableName;
+                        if(param.actualVariableName == null && variableCollisions.ContainsKey(paramRef))
+                        {
+                            s = CreateActualVariableName(param.desiredVariableName, 0);
+                        }
+                        if (!initialized.Contains(paramRef))
+                        {
+                            mainFunction.AddVariableDeclarationStatement(param.shaderType, s, param.initializationString);
+                            initialized.Add(paramRef);
+                        }
+                        paramStrings.Add(s);
+                        if(!inputs.Contains(paramRef) && param is ReferableData referable)
+                        {
+                            inputs.Add(paramRef);
+                            inputBuilder.AddField(referable.field);
+                        }
+                    }
+                    else
+                    {
+                        paramStrings.Add(param.initializationString);
+                    }
+
+                }
+                var function = functionReferences[functionRef];
+                mainFunction.CallFunction(function, paramStrings.ToArray());
+            }
+
+            private IEnumerable<VariableReference> RegisterVariables(NodeHandler workunit, FunctionReference? functionReference)
+            {
+                //if this is a run of the mill, function generating node. Generate local variables to handle passing around block-local values.
+                if (functionReference.HasValue)
+                {
+                    var function = functionReferences[functionReference.Value];
+                    foreach (var param in function.Parameters)
+                    {
+                        var port = workunit.GetPort(param.Name);
+                        if (param.IsInput)
+                        {
+                            //check and see if this port is already mapped to a variable - I dont think this should ever actually happen
+                            if (portToVariableLookup.TryGetValue(port.ID, out VariableReference varref))
```

Comment:
This seems to be checking for conflicts in inputs to a function, but its confusing cause its still doing work if the conflict is found? The comment adds some more ambiguity cause it indicates that this should never actually happen 🤔

Reply:
Yeah so thats a case I was talking to Brett about that I dont _think_ this case should ever get hit. I wanted to go ahead and write how we would handle the case while I was thinking about it, but assuming after full migration the debug message still never gets hit then I will likely remove checking for that case